src/qibolab/instruments/rfsoc.py

```diff
@@ -76,19 +76,30 @@ def convert_pulse(pulse: Pulse, qubits: dict[int, Qubit]) -> rfsoc.Pulse:
     return replace_pulse_shape(rfsoc_pulse, pulse.shape)
 
 
-def convert_frequency_sweeper(sweeper: rfsoc.Sweeper, sequence: PulseSequence, qubits: dict[int, Qubit]):
-    """Convert frequencies for `qibosoq.abstract.Sweeper` considering LO and HZ_TO_MHZ."""
+def convert_units_sweeper(sweeper: rfsoc.Sweeper, sequence: PulseSequence, qubits: dict[int, Qubit]):
+    """Convert units for `qibosoq.abstract.Sweeper` considering also LOs."""
     for idx, jdx in enumerate(sweeper.indexes):
-        if sweeper.parameter[idx] is rfsoc.Parameter.FREQUENCY:
+        parameter = sweeper.parameter[idx]
+        if parameter is rfsoc.Parameter.FREQUENCY:
             pulse = sequence[jdx]
             lo_frequency = pulse_lo_frequency(pulse, qubits)
 
             sweeper.starts[idx] = (sweeper.starts[idx] - lo_frequency) * HZ_TO_MHZ
             sweeper.stops[idx] = (sweeper.stops[idx] - lo_frequency) * HZ_TO_MHZ
+        elif parameter is rfsoc.Parameter.START:
+            sweeper.starts[idx] = sweeper.starts[idx] * NS_TO_US
+            sweeper.stops[idx] = sweeper.stops[idx] * NS_TO_US
+        elif parameter is rfsoc.Parameter.RELATIVE_PHASE:
+            sweeper.starts[idx] = np.degrees(sweeper.starts[idx])
+            sweeper.stops[idx] = np.degrees(sweeper.stops[idx])
```

Comment:
This switches on the sweeper type are happening multiple types. I start thinking that it would be nice to collect each sweeper type, and the related code, in a `class` variant, and the variants in an `enum`.

Here you could then invoke with:
```py
sweeper.starts[idx], sweeper.stops[idx] = Sweepers[parameter].convert_units(sweeper, sequence, qubits, jdx)
```
Too many arguments on the right-hand side, but we could put together `start, stop`, and do something like:
```py
pulse = sequence[jdx]
sweeper.ends[idx] = Sweepers[parameter].convert_units(sweeper.ends[idx], pulse=pulse, qubits=qubits)
```
to avoid passing the whole sequence (not needed in general, more isolated the better - and indexing is cheap), and to keep the extra arguments in `**kwargs` in the implementation (such that they are simply ignored by the others).

Reply:
Having multiple class variant would be nice and indeed here it would look much better, however I still have to convert everything to lists to send it through the socket and, when I re-create the object on the board, I need to know what kind of object they are. So I'm not sure this would work out of the box...

Reply:
I would use the classes just to collect the sweeper-specific code, not as the output. Then you use the classes in place of the switches, but the output will be the same :)